debitpolicy/debitpolicy.go

```diff
@@ -0,0 +1,115 @@
+package debitpolicy
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dogmatiq/dogma"
+	"github.com/dogmatiq/example/messages/commands"
+	"github.com/dogmatiq/example/messages/events"
+)
+
+// debitpolicy is the aggregate root for the debit policy.
+type debitpolicy struct {
+	// DebitAmount is the current sum of debit amounts for the period, in cents.
+	DebitAmount int64
+}
+
+func (p *debitpolicy) ApplyEvent(m dogma.Message) {
+	switch x := m.(type) {
+	case events.AccountDebitedForWithdrawal:
+		p.DebitAmount += x.Amount
+	case events.AccountDebitedForTransfer:
+		p.DebitAmount += x.Amount
+	}
+}
```

Comment:
The events references in this function will never be passed here.

`AggregateRoot.ApplyEvent()` is not a generic "event handler", it only receives events recorded from the aggregate message handler it's used with.

Reply:
Hmm ok, how will integration work across different systems/areas/(not sure what to call these)?

Reply:
It *has* to be the events you record within the debit policy message handler (`TransferApprovedByDebitPolicy`, etc).

Recall that we will need to decrement the counter if the debit ends up being declined by some other means (the compensating action in our "saga").